py/desispec/database/redshift.py

```diff
@@ -362,7 +362,7 @@ def load_file(filepath, tcls, hdu=1, expand=None, convert=None, index=None,
         return
     if maxrows == 0:
         maxrows = len(data)
-    log.info("Read data from %s.", filepath)
+    log.info("Read data from {} HDU {}.".format(filepath, hdu))
```

Comment:
Please don't change the formatting of the log messages.  There is a very specific reason for using `%`-style formatting.  See https://docs.python.org/3/howto/logging.html#logging-variable-data.  In theory other formatting methods are supported, but would require modifications to the `desiutil.log` module.  In addition, the logging infrastructure is smart enough to *not* format strings that it doesn't need to, as long as you are following the formatting style it is configured for.

Reply:
I don't understand this objection.  The new-style formatting is generating a string that gets passed to the logger, without the logger knowing or caring how that string was generated.  i.e. this doesn't require any modifications to `desiutil.log`; it is equivalent to:
```python
message = "Read data from {} HDU {}.".format(filepath, hdu)
log.info(message)
```

Reply:
The logger does know and does care.  Please read the link I sent in detail.

Reply:
I have empirically verified by running the code and looking at the log output that the formatting in this PR works just fine.  My understanding of that link you sent (and its links) is that the logger knows and cares only if you want the logger do the string formatting for you by passing in additional positional arguments.  If you format the string ahead of time before passing it to the logger (like I do here; note that there are no positional arguments or string parsing for the logger to do), then the logger doesn't know or care how you generated the string that you provide it.

For the record, I switched to the new-style formatting because I wanted to log the `hdu` but it can either be a string or an integer and I didn't realize until just now that old-style string formatting knows how to cast an integer to a string (it won't do the reverse).  But it just doesn't seem worth going back and changing working code to different working code.

Reply:
Contact me on Slack.  It is important that you understand this.